src/agent/src/image_rpc.rs

```diff
@@ -44,6 +70,184 @@ impl ImageService {
+        let aa_kbc_params = "kbc:ip:port".to_string();
+        if aa_kbc_params != "" {
+            let decrypt_config = format!("provider:attestation-agent:{}", aa_kbc_params);
+            cc = create_decrypt_config(vec![decrypt_config], vec![])?;
+        }
+
+        // Covert docker layer media type to OCI type
+        for layer_desc in image_manifest.layers.iter_mut() {
+            if layer_desc.media_type == manifest::IMAGE_DOCKER_LAYER_GZIP_MEDIA_TYPE {
+                layer_desc.media_type = manifest::IMAGE_LAYER_GZIP_MEDIA_TYPE.to_string();
+            }
+        }
+
+        for layer in image_data.layers.iter() {
+            let layer_digest = layer.clone().sha256_digest();
+
+            if layer.media_type == MEDIA_TYPE_LAYEE_GZIP_ENC
+                || layer.media_type == MEDIA_TYPE_LAYER_ENC
+            {
+                if cc.decrypt_config.is_none() {
+                    break;
+                }
+
+                for layer_desc in image_manifest.layers.iter_mut() {
+                    if layer_desc.digest == layer_digest {
+                        let (layer_decryptor, _dec_digest) = decrypt_layer(
+                            &cc.decrypt_config.as_ref().unwrap(),
+                            layer.data.as_slice(),
+                            layer_desc,
+                            false,
+                        )?;
+                        let mut plaintxt_data: Vec<u8> = Vec::new();
+                        let mut decryptor = layer_decryptor.unwrap();
+
+                        decryptor.read_to_end(&mut plaintxt_data)?;
+                        let layer_name = format!("{:x}", Sha256::digest(&plaintxt_data));
+                        let mut out_file =
+                            File::create(format!("{}/{}", oci_blob_path, layer_name))?;
+                        info!(sl!(), "Saving image file {}...", layer_name);
+                        out_file.write_all(&plaintxt_data)?;
+                        layer_desc.media_type = manifest::IMAGE_LAYER_GZIP_MEDIA_TYPE.to_string();
+
+                        layer_desc.digest = format!("sha256:{}", layer_name);
+                    }
+                }
+            } else if let Some(layer_name) = layer_digest.strip_prefix("sha256:") {
+                let mut out_file = File::create(format!("{}/{}", oci_blob_path, layer_name))?;
+                info!(sl!(), "Saving image file {}...", layer_name);
+                out_file.write_all(&layer.data)?;
+            }
+        }
+        let manifest_json = serde_json::to_string(&image_manifest)?;
+
+        let image_manifest_file = format!(
+            "{}{:x}",
+            oci_blob_path,
+            Sha256::digest(manifest_json.as_bytes())
+        );
+
+        let mut out_file = File::create(&image_manifest_file)?;
+        out_file.write_all(manifest_json.as_bytes())?;
+
+        let mut annotations = HashMap::new();
+        annotations.insert(
+            "org.opencontainers.image.ref.name".to_string(),
```

Comment:
Please make this a `const` with an explanatory comment.

Reply:
yes, it should be const, which not defined  by current oci-distribution crate, will fix it.